src/contexts/NewDelivery.tsx

```diff
@@ -0,0 +1,114 @@
+import { createContext, ReactNode, useState } from 'react'
+import { SubmitHandler } from 'react-hook-form'
+import useNewDeliverySteps from '@/hooks/useNewDeliverySteps'
+import { NewDeliveryContextValues } from '@/interfaces/newDeliveryContextValues'
+import useSetStep3Items from '@/hooks/useSetStep3Items'
+import useGenerateLeftItemsAlert from '@/hooks/useGenerateLeftItemsAlert'
+import { MoveGood } from '@/interfaces/moveGood'
+
+interface NewDeliveryProviderProps {
+  children: ReactNode
+}
+
+export const NewDeliveryContext = createContext<NewDeliveryContextValues>({
+  currentStep: 1,
+  formsData: {},
+  openDialog: false,
+  onCloseDialog: () => {},
+  handleBack: () => {},
+  handleClickOpen: () => {},
+  handleSubmit: () => {},
+  updateStep4Item: () => {},
+  deleteStep4Item: () => {},
+  alertMessage: [],
+  isCompletedMove: false,
+  isExceedQuantity: false,
+  step3Items: { pallets: 0, packages: 0, pieces: 0 },
+  setStep3Items: () => {}
+})
+
+export default function NewDeliveryProvider({ children }: NewDeliveryProviderProps) {
+  const [currentStep, setCurrentStep] = useState(1)
+  const [formsData, setFormsData] = useState<any>({})
+  const [openDialog, setOpenDialog] = useState(false)
+  const steps = useNewDeliverySteps()
+  const [step3Items, setStep3Items] = useState({ pallets: 0, packages: 0, pieces: 0 })
+  const [step4Items, setStep4Items] = useState<MoveGood[]>([{ type: '', quantity: 0, zone: '' }])
```

Comment:
lets have these renamed as we talked over chat.

Reply:
Fixed